src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,128 @@
+                    return Optional.of(matToBook(resultSet));
+                }
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format("Getting book by id=%d was failed", id), e
+            );
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_QUERY);
+                ResultSet resultSet = statement.executeQuery()) {
+            List<Book> books = new ArrayList<>();
+            while (resultSet.next()) {
+                books.add(matToBook(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Retrieving all books from DB was failed", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_QUERY)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException(
+                        String.format("Book with id=%d wasn't updated, no rows affected.",
+                                book.getId())
+                );
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format("Updating book with id=%d was failed", book.getId()), e
+            );
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(DELETE_QUERY)) {
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format("Deletion book by id=%d was failed", id), e
+            );
+        }
+    }
+
+    private Long getCreatedId(PreparedStatement statement) throws SQLException {
+        int affectedRows = statement.executeUpdate();
+        if (affectedRows < 1) {
+            throw new DataProcessingException("Creating book failed, no rows affected.");
+        }
+        try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+            if (!generatedKeys.next()) {
+                throw new DataProcessingException("Creating book failed, no ID obtained.");
+            }
+            return generatedKeys.getLong(1);
+        }
+    }
```

Comment:
this method has a hidden behavior - `statement.executeUpdate`

this is bad, since it is not obvious for the caller that this method is going to execute some query against DB

basically this method should simply retrieve the generated key and that's all